recipes/soci/all/conanfile.py

```diff
@@ -0,0 +1,170 @@
+                  path,
+                  lib_filename
+                )
+                if os.path.isfile(full_path):
+                    return full_path
+            print('{} not found in {}'.format(lib_filename, self.deps_cpp_info[name].lib_paths))
+            return None
+
+        libs = [find_lib_path(name, lib) for lib in self.deps_cpp_info[name].libs]
+
+        return namedtuple('Dependency', ['root', 'libs'])(
+          root          = self.deps_cpp_info[name].rootpath,
+          libs          = libs
+        )
+
+    def _configure_cmake(self):
+        cmake = CMake(self)
+        cmake.definitions["SOCI_SHARED"] = self.options.shared
+        cmake.definitions["SOCI_STATIC"] = not self.options.shared
+        cmake.definitions["SOCI_TESTS"] = False
+
+        print("self.settings.compiler.cppstd = {}".format(self.settings.compiler.cppstd))
+        if self.settings.compiler.cppstd != 'None':
+            cmake.definitions["CMAKE_CXX_STANDARD"] = self.settings.compiler.cppstd
+
+        cmake.definitions["WITH_EMPTY"] = self._with_backend(self.options.with_backend_empty)
+
+        if self._with_backend(self.options.with_backend_sqlite3):
+            sqlite3 = self._dependency('sqlite3')
+            cmake.definitions["WITH_SQLITE3"] = "ON"
+            cmake.definitions["SQLITE_ROOT_DIR"] = sqlite3.root
+
+        if self._with_backend(self.options.with_backend_mysql):
+            mysql = self._dependency('libmysqlclient')
+            cmake.definitions["WITH_MYSQL"] = "ON"
+            os.environ["MYSQL_DIR"] = mysql.root
+            cmake.definitions["MYSQL_LIBRARIES"] = ';'.join(mysql.libs)
+
+        if self._with_backend(self.options.with_backend_odbc):
+            odbc = self._dependency('odbc')
+            cmake.definitions["WITH_ODBC"] = "ON"
+            cmake.definitions["ODBC_INCLUDE_DIR"] = os.path.join(odbc.root, 'include')
+            if self.settings.os == 'Windows':
+              cmake.definitions["ODBC_LIBRARY"] = os.path.join(odbc.root, 'lib', 'libodbc.lib')
+            else:
+              cmake.definitions["SOCI_ODBC_DO_NOT_TEST"] = True
+              if self.options['odbc'].shared == False:
+                cmake.definitions["ODBC_LIBRARY"] = os.path.join(odbc.root, 'lib', 'libodbc.a')
+              else:
+                cmake.definitions["ODBC_LIBRARY"] = os.path.join(odbc.root, 'lib', 'libodbc.so')
+
+
+        if self._with_backend(self.options.with_backend_postgresql):
+            postgresql = self._dependency('libpq')
+            cmake.definitions["WITH_POSTGRESQL"] = "ON"
+            os.environ["POSTGRESQL_ROOT"] = postgresql.root
+
+        #cmake.configure(source_folder=self._source_subfolder)
+        cmake.configure()
+        cmake.verbose = True
+        return cmake
+
+    def build(self):
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def package_info(self):
+        self.cpp_info.build_modules.append('cmake/SOCI-override.cmake')
```

Comment:
By using components, you don't need to supply your own cmake script.
As I've written above, the cmake part of component is not yet available.